### PR TITLE
Add support to Redirect with http code.

### PR DESCRIPTION
**Pull request by SARAVANA1501**
Summary of the changes (Less than 80 chars)

- Add new constructor to accept HTTP redirection code

- Added code to handle custom code in Redirection context Executor

Addresses #13359

### CODE REVIEW

**Comment by SARAVANA1501** [comment]
Please review and provide your comments.

**Comment by mkArtakMSFT** [comment]
Thanks for your PR, @SARAVANA1501.
@pranavkm can you please review this and merge if it looks good? Thanks!

File: src/Mvc/Mvc.Core/src/RedirectResult.cs [hunk 1]
```diff
@@ -94,6 +123,14 @@ public string Url
             }
         }
 
+        /// <summary>
+        /// Gets the status code to redirect with.
+        /// </summary>
+        public int? statusCode
```
Comment (mkArtakMSFT) [inline src/Mvc/Mvc.Core/src/RedirectResult.cs]: Please use Pascal Case for properties: `StatusCode` with capital letter

Comment (mkArtakMSFT) [reply src/Mvc/Mvc.Core/src/RedirectResult.cs]: Also you can rewrite this as `public int? StatusCode => _statusCode;`

Comment (SARAVANA1501) [reply src/Mvc/Mvc.Core/src/RedirectResult.cs]: Modified  as requested.

**Comment by mkArtakMSFT** [comment]
Thanks for your effort, @SARAVANA1501.
After another review of the issue we've decided not to make any changes to the framework and instead encourage customers to use the `StatusCodeResult(303)` instead.

**Pull request closed without merging**